Api/Services/EventService.cs

```diff
@@ -0,0 +1,156 @@
+                CreatorFullName = user.FullName,
+                RestaurantId = newEvent.RestaurantId,
+                RestaurantName = newEvent.Restaurant.Name,
+                VisitId = newEvent.VisitId,
+                Interested = newEvent.Interested.Select(i => new UserSummaryVM
+                {
+                    FirstName = i.FirstName,
+                    LastName = i.LastName,
+                    UserId = i.Id
+                }).ToList()
+            };
+        }
+
+        public async Task<Result<EventVM>> GetEventAsync(int id)
+        {
+            var checkedEvent = await context.Events
+                .Include(e => e.Interested)
+                .Include(e => e.Creator)
+                .Include(e => e.Restaurant)
+                .FirstOrDefaultAsync(e => e.Id == id);
+            if (checkedEvent is null)
+            {
+                return new ValidationFailure
+                {
+                    PropertyName = null,
+                    ErrorCode = ErrorCodes.NotFound,
+                    ErrorMessage = ErrorCodes.NotFound
+                };
+            }
+            return new EventVM
+            {
+                CreatedAt = checkedEvent.CreatedAt,
+                Description = checkedEvent.Description,
+                Time = checkedEvent.Time,
+                Id = checkedEvent.Id,
+                MustJoinUntil = checkedEvent.MustJoinUntil,
+                CreatorId = checkedEvent.CreatorId,
+                CreatorFullName = checkedEvent.Creator.FullName,
+                RestaurantId = checkedEvent.RestaurantId,
+                RestaurantName = checkedEvent.Restaurant.Name,
+                VisitId = checkedEvent.VisitId,
+                Interested = checkedEvent.Interested.Select(i => new UserSummaryVM
+                {
+                    FirstName = i.FirstName,
+                    LastName = i.LastName,
+                    UserId = i.Id
+                }).ToList()
+            };
+        }
+        public async Task<Result<List<EventSummaryVM>>> GetEventsCreatedAsync(User user)
+        {
+            var events = await context.Events
+                .Include(e => e.Interested)
+                .Include(e => e.Creator)
+                .Include(e => e.Restaurant)
+                .Where(e => e.CreatorId == user.Id)
+                .ToListAsync();
+
+            if (events is null)
+            {
+                return new ValidationFailure
+                {
+                    PropertyName = null,
+                    ErrorCode = ErrorCodes.NotFound,
+                    ErrorMessage = ErrorCodes.NotFound
+                };
+            }
+            if (events.Count == 0)
+            {
+                return new ValidationFailure
+                {
+                    PropertyName = null,
+                    ErrorCode = ErrorCodes.NotFound,
+                    ErrorMessage = ErrorCodes.NotFound
+                };
+            }
```

Comment:
To nie jest błąd jeśli lista jest pusta

Reply:
Czy mimo wszystko nie lepiej będzie poinformować front o tym, że nie znaleziono żadnych pasujących eventów, niż zwrócić pustą listę, żeby się zastanawiali co jest nie tak, czemu nic im się nie wyświetla?